06_gpu/stable_diffusion_cli.py

```diff
@@ -0,0 +1,118 @@
+# ---
+# output-directory: "/tmp/stable-diffusion"
+# ---
+# # Stable Diffusion CLI
+#
+# This tutorial shows how you can create a CLI tool that runs GPU-intensive
+# work remotely but feels like you are running locally. We will be building
+# a tool that generates an image based on a prompt against Stable Diffusion
+# using the HuggingFace Hub and the `diffusers` library.
+
+# ## Basic setup
+import modal
+from pathlib import Path
+
+# All Modal programs need a [`Stub`](/docs/reference/modal.Stub) — an object that acts as a recipe for
+# the application. Let's give it a friendly name.
+
+stub = modal.Stub("stable-diffusion-cli")
+
+# We will be using `typer` to create our CLI interface.
+
+import typer
+
+app = typer.Typer()
+
+# ## Model dependencies
+#
+# Your model will be running remotely inside a container. We will be installing
+# all the model dependencies in the next step. We will also be "baking the model"
+# into the image using the script `download_stable_diffusion_models.py`.
+# This is technique that allows you to copy model files to
+# a worker more efficiently because they only need to be moved once.
+
+image = (
+    modal.Image.conda()
```

Comment:
Not super important but is it worth doing this without conda? Since conda is kind of slow and harder to work with (imo)

Reply:
I had a version without conda but I am not able to compile xformers without it yet.

Inference is about 20% slower without xformers installed.